pkg/controller/storagecluster/update.go

```diff
@@ -332,6 +337,70 @@ func (c *Controller) getKVDBNodeAvailability(cluster *corev1.StorageCluster) (in
 
 }
 
+func (c *Controller) parallelUpgradeNodesList(cluster *corev1.StorageCluster, pods []*v1.Pod) []*v1.Pod {
+	if len(pods) == 0 {
+		return pods
+	}
+
+	if storagePodsEnabled(cluster) {
+		storageNodeList, err := c.Driver.GetStorageNodes(cluster)
```

Comment:
We call `Driver.GetStorageNodes` in `getPodAvailability` method as well. Instead let's call it only once in rollingUpdate and then pass the output list to both these methods.